pydicer/convert/convert_pt.py

```diff
@@ -0,0 +1,311 @@
+            "%Y%m%d%H%M%S",
+        )
+        serie_datetime = datetime.strptime(
+            s[Tag(0x00080021)].value + s[Tag(0x00080031)].value.split(".")[0],
+            "%Y%m%d%H%M%S",
+        )
+
+        try:
+            if datetime(1950, 1, 1) < serie_datetime <= acquisition_datetime:
+                scan_datetime = serie_datetime
+            else:
+                scan_datetime_value = s[Tag(0x0009100D)].value
+                if isinstance(scan_datetime_value, bytes):
+                    scan_datetime_str = scan_datetime_value.decode("utf-8").split(".")[
+                        0
+                    ]
+                elif isinstance(scan_datetime_value, str):
+                    scan_datetime_str = scan_datetime_value.split(".")[0]
+                else:
+                    raise ValueError("The value of scandatetime is not handled")
+                scan_datetime = datetime.strptime(scan_datetime_str, "%Y%m%d%H%M%S")
+
+            start_time_str = s.RadiopharmaceuticalInformationSequence[
+                0
+            ].RadiopharmaceuticalStartTime
+            start_time = time(
+                int(start_time_str[0:2]),
+                int(start_time_str[2:4]),
+                int(start_time_str[4:6]),
+            )
+            start_datetime = datetime.combine(scan_datetime.date(), start_time)
+            decay_time = (scan_datetime - start_datetime).total_seconds()
+        except KeyError:
+            warnings.warn(
+                "Estimation of time decay for SUV"
+                " computation from average parameters"
+            )
+            decay_time = 1.75 * 3600  # From Martin's code
+        except AttributeError:
+            warnings.warn(
+                "'Dataset' object has no attribute 'RadiopharmaceuticalStartTime'"
+            )
+            decay_time = 1.75 * 3600  # From Shuchao's code
+        suv_results = get_suv_from_bqml(slices, decay_time, patient_weight, dtype=dtype)
+
+    elif units == "CNTS":
+        suv_results = get_suv_philips(slices, dtype=dtype)
+    else:
+        raise ValueError("The {} units is not handled".format(units))
+
+    return suv_results
+
+
+def get_suv_philips(slices, dtype=np.float32):
+    """Function to Get SUV from raw PET if units = "CNTS"
+    Args:
+    Returns:
+    """
+    image = list()
+    suv_scale_factor_tag = Tag(0x70531000)
+    for s in slices:
+        if (
+                suv_scale_factor_tag in s
+                and "RescaleSlope" in s
+                and "RescaleIntercept" in s
+            ):  # # From Shuchao's code
+            im = (
+                float(s.RescaleSlope) * s.pixel_array + float(s.RescaleIntercept)
+            ) * float(s[suv_scale_factor_tag].value)
+        else:
+            im = (1.0 * s.pixel_array + 0.0) * 0.000587
+
+        image.append(im)
+
+    return np.stack(image, axis=-1).astype(dtype)
+
+
+def get_suv_from_bqml(slices, decay_time, patient_weight, dtype=np.float32):

```

Comment:
Please add more elaborate docstring in here, also is there a need for the floating point datatype to be configurable?